Mathlib/Analysis/Convex/Cone/Pointed.lean

```diff
@@ -0,0 +1,191 @@
+/-
+Copyright (c) 2023 Apurva Nakade All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Apurva Nakade
+-/
+import Mathlib.Analysis.Convex.Cone.Dual
+import Mathlib.Algebra.Order.Nonneg.Module
+import Mathlib.Algebra.Module.Submodule.Basic
+
+
+/-!
+# Pointed cones
+
+A *pointed cone* is defined to be a submodule of a module where the scalars are restricted to be
+nonnegative. This is equivalent to saying that as a set a pointed cone is convex cone which
+contains `0`. This is a bundled version of `ConvexCone.Pointed`. We choose the submodule definition
+as it allows us to use the `Module` API to work with convex cones.
+
+## TODO
+
+- Rewrite proper cones using pointed cones.
+- Construct products and/or direct sums of pointed cones.
+
+-/
+
+variable {𝕜 E F G : Type*}
+
+/-- A pointed cone is a submodule of a module with scalars restricted to being nonnegative. -/
+abbrev PointedCone (𝕜 : Type*) (E : Type*) [OrderedSemiring 𝕜] [AddCommMonoid E] [Module 𝕜 E] :=
+  have : Module { c : 𝕜 // 0 ≤ c } E := Module.compHom E (@Nonneg.coeRingHom 𝕜 _)
+  Submodule { c : 𝕜 // 0 ≤ c } E
+
+namespace PointedCone
+
+set_option quotPrecheck false in
+/-- The set of non-negative elements. -/
+scoped notation "𝕜≥0" => { c : 𝕜 // 0 ≤ c }
```

Comment:
Why not just `open scoped Nonneg`?

Reply:
I tried this but I keep getting the error 

```lean
unknown constant '𝕜✝'
```

Reply:
Then your notation is broken :(. What error do you get when you remove the `set_option`?

Reply:
Without the `set_option` the error is 

```lean
unknown identifier '𝕜' at quotation precheck; you can use `set_option quotPrecheck false` to disable this check.
```

I'll just remove this notation completely. I think there should be a notation introduced much earlier in the `Nonneg.Ring` files.

Reply:
Ah yeah, your notation is bogus. You're making `𝕜≥0` be notation for `{a : α // 0 ≤ a}` for *any* `α`. My naïve attempt `
notation α "≥0" => {a : α // 0 ≤ a}` fails too, so I'm asking @kmill.

Reply:
You need `local notation` (and be sure to drop `scoped`). In principle having a notation with a local `variable` is fine, but it only works within the section it's defined. Be aware that at the moment you should not use `𝕜≥0` in further `variable` commands (this issue is tracked [here](https://github.com/leanprover/lean4/issues/2535)).

I know `notation3` supports local variables, though allowing it to pretty print would need #6833.

If `local notation` doesn't work, try
```lean
local notation3 (prettyPrint := false) "𝕜≥0" => { c : 𝕜 // 0 ≤ c }
```
Once #6833 is in you should be able to delete `(prettyPrint := false)`.